hack/builder/version.sh

```diff
@@ -1,3 +1,3 @@
 VERSION=$(date +"%y%m%d%H%M")-$(git rev-parse --short HEAD)
 # TODO: reenable ppc64le when new builds are available
-ARCHITECTURES="amd64 arm64v8"
+ARCHITECTURES="amd64"
```

Comment:
@rmohr What do you think about the idea of building the builder image on top of the libvirt-devel image defined via bazeldnf? I actually don't see that image used anywhere at the moment, so I think that maybe it was introduced specifically for this purpose and you just never got around to actually using it?

Reply:
Regarding to removing `arm64v8` here: I think having an arm64 builder image is not relevant anymore. @zhlhahaha and I spent quite some time on full cross-compile support (and that is alwo what we do in CI and on arm releases). @zhlhahaha what do you think?

Regarding

> I actually don't see that image used anywhere at the moment

I am not sure what you mean. But right now, if you would develop directly on an arm machine and you would run `make` it would be used.

Reply:
Sorry, what I meant to say is that it doesn't look like the `libvirt-devel` image defined (along with `launcherbase`) in `hack/rpm-deps.sh` / `rpm/BUILD.bazel` is used anywhere. Could we use that as the base for the builder image? That way we would also get the advantage of full reproducibility, at least for the RPM part.

Regarding cross-compilation, of course the fact that you can do that seamlessly is great, especially for CI, but can we really tell people that they should just always develop on x86_64? That feels wrong...

Reply:
> Sorry, what I meant to say is that it doesn't look like the `libvirt-devel` image defined (along with `launcherbase`) in `hack/rpm-deps.sh` / `rpm/BUILD.bazel` is used anywhere. Could we use that as the base for the builder image? That way we would also get the advantage of full reproducibility, at least for the RPM part.

Now I understand. The `libvirt-devel` rpmtree is the base for the `libvirt-libs_aarch64` and `libvirt-libs_x86_64` targets. They in turn are used to provide headers and shared libraries to this target:

```python
cc_library(
    name = "libvirt-libs",
    srcs = select({
        "@io_bazel_rules_go//go/platform:linux_arm64": ["//rpm:libvirt-libs_aarch64/usr/lib64"],
        "//conditions:default": ["//rpm:libvirt-libs_x86_64/usr/lib64"],
    }),
    hdrs = select({
        "@io_bazel_rules_go//go/platform:linux_arm64": ["//rpm:libvirt-libs_aarch64/usr/include/libvirt"],
        "//conditions:default": ["//rpm:libvirt-libs_x86_64/usr/include/libvirt"],
    }),
    include_prefix = "libvirt",
    linkstatic = 1,
    strip_include_prefix = select({
        "@io_bazel_rules_go//go/platform:linux_arm64": "/rpm/libvirt-libs_aarch64/",
        "//conditions:default": "/rpm/libvirt-libs_x86_64/",
    }),
    visibility = ["//visibility:public"],
)
```

`libvirt-devel` is only kept in the builder image to allow the pure go builds to work. Bazel does not use it anymore. :)

> Regarding cross-compilation, of course the fact that you can do that seamlessly is great, especially for CI, but can we really tell people that they should just always develop on x86_64? That feels wrong...

I have no issue with keeping it. But if you have difficulties keeping it, I would be fine with removing it. One can also build with bazel on arm without the builder image (only `make generate` would be a pain due to the external pre-installed dependencies). I guess arm as developer workstation may become more important with recent switches from some companies.

Reply:
Can we keep building multi-arch builder image? It would be much helpful for people who develop kubevirt on ARM. Only a few codes need to be added and I will put them in the comments.

Reply:
> The `libvirt-devel` rpmtree is the base for the `libvirt-libs_aarch64` and `libvirt-libs_x86_64` targets.

Oh, I had missed that connection. Though I've been getting better at understanding Bazel rules, as you can clearly see it still doesn't come quite naturally to me :)

> I have no issue with keeping it. But if you have difficulties keeping it, I would be fine with removing it.

I would definitely prefer keeping it if at all possible, because I believe that you should be able to use the same steps to build and test on any supported architecture and I certainly don't want to be the one who introduces regressions in this area :)

So, do you think it would be possible to reuse the existing `libvirt-devel` image (or a similar image built using Bazel with RPM content controlled by bazeldnf) as the base for the builder image?

If I'm looking at https://github.com/bazelbuild/rules_docker/tree/master/testing/examples/run_instruction_arbitrary correctly it should be possible to take the existing Dockerfile and convert it to a bunch of `container_run_and_commit()` rules, but how would you then get the image "out" of Bazel in order to assemble a manifest (something that `rules_docker` currently can't do AFAICT) and push the result to a registry?

Reply:
> So, do you think it would be possible to reuse the existing `libvirt-devel` image (or a similar image built using Bazel with RPM content controlled by bazeldnf) as the base for the builder image?

It is possible the way you describe above. But I would prefer for now to keep them separated.

Reply:
Okay! Thanks to @zhlhahaha's suggestions below, I think I have managed to make it work correctly with only minimal changes to the current process, so I no longer have pressing motives to turning the way we build that container image upside down :)